server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -3099,25 +3099,24 @@ public void test_subscript_on_aliased_json_string_cast_to_object() throws Except
 
         // DYNAMIC object cast
         executor.getSessionSettings().setErrorOnUnknownObjectKey(true);
-        // If not wrapped into a sub-select, the JSON string expression gets normalized and results in a typed object literal
-        var analyzed = executor.analyze("SELECT '{\"a\":1}'::OBJECT['a']");
-        assertThat(analyzed.outputs().getFirst()).isLiteral(1, DataTypes.INTEGER);
-        // Normalization is not possible on references, such the return type is unknown
-        analyzed = executor.analyze("SELECT js::OBJECT['a'] FROM t1");
-        assertThat(analyzed.outputs().getFirst()).isFunction("subscript", DataTypes.UNDEFINED);
-
-        // Once wrapped, the subscript function gets resolved before any normalization, resulting in an exception
+        // Subscript on literals
+        assertThatThrownBy(() -> executor.analyze("SELECT '{\"a\":1}'::OBJECT['a']"))
+            .isExactlyInstanceOf(ColumnUnknownException.class)
+            .hasMessageContaining("The expression's return type `OBJECT(DYNAMIC)` does not contain the key `a`");
```

Comment:
The third commit looks good to me. BTW, should we mention `::object` in breaking changes as well? Now that it results in columnless objects.